go/libraries/doltcore/sqle/table_editor_fk_test.go

```diff
@@ -0,0 +1,479 @@
+				var err error
+				root, err = executeModify(context.Background(), fk_dEnv, root, sqlStatement)
+				require.NoError(t, err)
+			}
+
+			t.Run("one", func(t *testing.T) {
+				testTableEditorRows(t, root, test.expectedOne, "one")
+			})
+			t.Run("two", func(t *testing.T) {
+				testTableEditorRows(t, root, test.expectedTwo, "two")
+			})
+		})
+	}
+}
+
+func TestTableEditorForeignKeyRestrict(t *testing.T) {
+	for _, referenceOption := range []string{
+		"ON DELETE RESTRICT ON UPDATE RESTRICT",
+		"ON DELETE NO ACTION ON UPDATE NO ACTION",
+		"",
+	} {
+		t.Run(referenceOption, func(t *testing.T) {
+			testRoot, err := ExecuteSql(fk_dEnv, fk_initialRoot, fmt.Sprintf(`
+ALTER TABLE two ADD FOREIGN KEY (v1) REFERENCES one(v1) %s;
+INSERT INTO one VALUES (1, 1, 1), (2, 2, 2), (3, 3, 3);
+INSERT INTO two VALUES (1, 1, 1), (2, 2, 2), (3, 3, 3);
+`, referenceOption))
+			require.NoError(t, err)
+
+			tests := []struct {
+				setup       string
+				trigger     string
+				expectedErr bool
+			}{
+				{
+					``,
+					`UPDATE one SET v1 = v1;`,
+					false,
+				},
+				{
+					``,
+					`UPDATE one SET v2 = v1 + v2;`,
+					false,
+				},
+				{
+					``,
+					`UPDATE one SET v1 = v1 + v2;`,
+					true,
+				},
+				{
+					`
+INSERT INTO one VALUES (4, 4, 4);
+`,
+					`
+UPDATE one SET v1 = 1 WHERE pk = 4;
+`,
+					false,
+				},
+				{
+					`
+INSERT INTO one VALUES (4, 4, 4);
+`,
+					`
+DELETE FROM one WHERE pk > 3;
+`,
+					false,
+				},
+				{
+					`
+INSERT INTO one VALUES (4, 3, 4);
+`,
+					`
+DELETE FROM one WHERE pk > 3;
+`,
+					true,
+				},
+				{
+					`
+INSERT INTO one VALUES (4, 4, 4);
+UPDATE one SET v1 = 2 WHERE pk > 3;
+`,
+					`
+DELETE FROM one WHERE pk > 3;
+`,
+					true,
+				},
+				{
+					`
+INSERT INTO one VALUES (4, 4, 4);
+DELETE FROM two WHERE pk = 2;
+UPDATE one SET v1 = 2 WHERE pk > 3;
+`,
+					`
+DELETE FROM one WHERE pk > 3;
+`,
+					false,
+				},
+			}
+
+			for _, test := range tests {
+				t.Run(test.setup+test.trigger, func(t *testing.T) {
+					root := testRoot
+					for _, sqlStatement := range strings.Split(test.setup, ";") {
+						var err error
+						root, err = executeModify(context.Background(), fk_dEnv, root, sqlStatement)
+						require.NoError(t, err)
+					}
+					if test.expectedErr {
+						root, err = executeModify(context.Background(), fk_dEnv, root, test.trigger)
+						assert.Error(t, err)
+					} else {
+						root, err = executeModify(context.Background(), fk_dEnv, root, test.trigger)
+						assert.NoError(t, err)
+					}
+				})
+			}
+		})
+	}
+}
+
+func TestTableEditorForeignKeyViolations(t *testing.T) {
+	testRoot, err := ExecuteSql(fk_dEnv, fk_initialRoot, `
+ALTER TABLE two ADD FOREIGN KEY (v1) REFERENCES one(v1) ON DELETE CASCADE ON UPDATE CASCADE;
+ALTER TABLE three ADD FOREIGN KEY (v1, v2) REFERENCES two(v1, v2) ON DELETE CASCADE ON UPDATE CASCADE;
+`)
+	require.NoError(t, err)
+
+	tests := []struct {
```

Comment:
Formatting for these and other test cases in this file are a bit weird. I would do two things: 1) give a name to the anonymous struct to reduce one level of nesting, 2) standardize on for indentation of these strings. Consider adding named struct keys (GoLand will do this for you).

It's not a huge deal, but the formatting does make some of these a bit harder to read than they should be